cmd/clusterctl/hack/create-local-repository.py

```diff
@@ -126,7 +126,7 @@ def write_local_repository(provider, version, components_file, components_yaml):
             if e.errno != errno.EEXIST:
                 raise
         components_path = os.path.join(provider_folder, components_file)
-        f = open(components_path, 'wb')
+        f = open(components_path, 'w')
```

Comment:
I'm running into:

Traceback (most recent call last):
  File "cmd/clusterctl/hack/create-local-repository.py", line 130, in write_local_repository
    f.write(components_yaml)
TypeError: write() argument must be str, not bytes

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "cmd/clusterctl/hack/create-local-repository.py", line 241, in <module>
    repos = list(create_local_repositories())
  File "cmd/clusterctl/hack/create-local-repository.py", line 162, in create_local_repositories
    components_path = write_local_repository(provider, next_version, components_file, components_yaml)
  File "cmd/clusterctl/hack/create-local-repository.py", line 138, in write_local_repository
    raise Exception('failed to write {} to {}: {}'.format(components_file, provider_folder, e))
Exception: failed to write core-components.yaml to /home/****/.cluster-api/dev-repository/cluster-api/v0.3.8: write() argument must be str, not bytes

using 'wb' fixes the issue.